src/stub_launcher.h

```diff
@@ -151,6 +151,11 @@ class StubLauncher {
   TRITONSERVER_Error* ReceiveMessageFromStub(
       bi::managed_external_buffer::handle_t& message);
 
+#ifdef TRITON_ENABLE_GPU
+  // Share CUDA memory pool with stub process
+  void ShareCUDAMemoryPool(TRITONBACKEND_MemoryManager* triton_mem_manager);
```

Comment:
Switched to use the [`parent_to_stub_mq_`](https://github.com/triton-inference-server/python_backend/pull/293/files#diff-f2bb64cbc9d4dc5d938ce965813b23178f53732e02e8c3a2afa707e287e92e57R1085) instead of the original message queue to enable flexible sharing of the CUDA memory pool with the stub for all cases:
- [BLS request has GPU input](https://github.com/triton-inference-server/python_backend/pull/293/files#diff-ca67ec8693bbdfbd82a858e307f87fa39fc9187d39228f1afc5375837dc3f60dR698)
- [BLS response has GPU output](https://github.com/triton-inference-server/python_backend/pull/293/files#diff-ca67ec8693bbdfbd82a858e307f87fa39fc9187d39228f1afc5375837dc3f60dR1683)
- [Model has GPU input](https://github.com/triton-inference-server/python_backend/pull/293/files#diff-ca67ec8693bbdfbd82a858e307f87fa39fc9187d39228f1afc5375837dc3f60dR522)
- [Model has GPU output](https://github.com/triton-inference-server/python_backend/pull/293/files#diff-ca67ec8693bbdfbd82a858e307f87fa39fc9187d39228f1afc5375837dc3f60dR1552), [decoupled](https://github.com/triton-inference-server/python_backend/pull/293/files#diff-ca67ec8693bbdfbd82a858e307f87fa39fc9187d39228f1afc5375837dc3f60dR1169)